healthcheck/core.go

```diff
@@ -352,17 +359,22 @@ func (hc *Check) Notify() {
 // execute invokes the given healthcheck checker with the configured timeout.
 func (hc *Check) execute() *Result {
 	ch := make(chan *Result, 1)
+
+	hc.lock.RLock()
```

Comment:
I don't see the lock is needed. Everything is run by a single go routine expect for the checker.Check() below. So I think you can simply copy the checker and timeout without the lock?

Reply:
hc.Run is its own goroutine, while hc.Config could be updated from outside (Server.manager()), which is causing a data race. Actually I noticed this with `go test -race ./...`. And this PR is for the fix.

Reply:
could you show me where Server.manager() is modifying hc.Config? I only see it's piping through the channel.

Reply:
Sorry, I take that one back. It's not Server.manager() causing a race with hc.Config, but the goroutine fired from Check.execute() [1], in which hc.timeout value is being read while the same could be updated from hc.Run goroutine.

[1] https://github.com/google/seesaw/blob/master/healthcheck/core.go#L359

Reply:
Sorry, I take that one back. It's actually goroutine fired from Check.execute() itself [1] causing a data race with the Check.Run() goroutine. In the former one, hc.timeout is being read while it could be updated in the latter one.

[1] https://github.com/google/seesaw/blob/master/healthcheck/core.go#L359

Reply:
right. that's why I was saying we don't need the lock but just coping the timeout and checker in the execute() before spawning the goroutine

Reply:
I see what you're saying now. Just updated another version your way. Commit message also updated to reflect the same.

Besides, I just found the other one you just merged #80 has a problem on its own. Hence added another commit in this PR to address that. Hope you don't mind. These two together gives `go test -race ./...` a clean pass.

BTW, did this repository enable Github Action? If yes, I can draft another PR to enable Github Action on seesaw to run unit tests. So that regression can be captured earlier in pull request stage.

Reply:
Go ahead with Github Action. Thanks for the initiative.